dataset/src/main/kotlin/com/cosmotech/dataset/service/DatasetServiceImpl.kt

```diff
@@ -278,8 +278,9 @@ class DatasetServiceImpl(
       record?.values()?.forEach { element ->
         when (element) {
           is Node -> {
-            val csmGraphEntity = element.toCsmGraphEntity(CsmGraphEntityType.NODE)
```

Comment:
You rather fix the toCsmGraphEntity function. This function does more than just getProperty.